### PR TITLE
BCDA-2460 Feature: Add Swagger documentation for admin endpoints

### PR DESCRIPTION
**Pull request by dhgreene**
### Fixes [BCDA-2460](https://jira.cms.gov/browse/BCDA-2460)
The team currently integrating with the SSAS admin endpoints has requested Swagger documentation.  This PR adds Swagger for the admin endpoints.

### Proposed Changes
- Add Swagger UI artifacts
- Add structs, comments, etc. for documenting the API in Swagger
- Add `documentation` Docker image
- Update error messages to conform with JSON spec
- Add Swagger to admin router

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

Adding Swagger documentation to the project should not present a significant new attack surface: the queries still go to the existing API protected by existing credentials and network security groups, and the examples already exist in other forms in the public repository.

### Acceptance Validation
Screen Shot 2019-12-09 at 5 16 51 PM
Screen Shot 2019-12-09 at 5 17 27 PM
Screen Shot 2019-12-09 at 5 18 09 PM
Screen Shot 2019-12-09 at 5 19 32 PM
Screen Shot 2019-12-09 at 5 21 21 PM

### Feedback Requested
* Play with Swagger
    * `docker-compose up documentation`
    * `docker-compose up ssas`
    * Get credentials from `make smoke-test`
    * [http://localhost:3104/swagger](http://localhost:3104/swagger)
* Creating separate structs to document the API inputs/outputs for Swagger constitutes duplication of code.  Should we take the time now to refactor for less duplication?
* Is the content in the Swagger service description good?  Are there better descriptions or examples for any individual data elements?

### CODE REVIEW

File: ssas/SwaggerStructs.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+// swagger:response publicKeyResponse
+type PublicKeyResponse struct {
+	// Public key details
+	// in: body
+	Body struct {
+		// This system's client ID
+		// Required: true
+		ClientID	string	`json:"client_id"`
+		// The public key (if any) registered for this system
+		// Required: true
```
Comment (tbellj) [inline ssas/SwaggerStructs.go]: can this be required if the public key is not guaranteed to be there?

File: ops/build_and_package.sh [hunk 2]
```diff
@@ -23,12 +23,19 @@ fi
...
+  exit 1
+fi
+
 cd ../ssas
 go clean
 echo "Building ssas..."
 go build -ldflags "-X github.com/CMSgov/bcda-ssas-app/ssas/constants.Version=$VERSION" -o ssas ./service/main
 echo "Packaging ssas binary into RPM..."
-fpm -v $VERSION -s dir -t rpm -n ssas ssas=/usr/local/bin/ssas
+fpm -v $VERSION -s dir -t rpm -n ssas ssas=/usr/local/bin/ssas swaggerui=/etc/sv/api _site=/etc/sv/api
```
Comment (msnook) [inline ops/build_and_package.sh]: Is `_site` extraneous here?

Comment (msnook) [inline ops/build_and_package.sh]: Should this be packaged in `/etc/sv/ssas` instead of `/etc/sv/api`?

File: ssas/SwaggerStructs.go (hunk 1 above)
Comment (dhgreene) [reply ssas/SwaggerStructs.go]: Yes!  If there's no public key, the response looks like:
  `{"client_id":"this_client_id","public_key":""}`

File: ops/build_and_package.sh (hunk 2 above)
Comment (dhgreene) [reply ops/build_and_package.sh]: Yep!  Removed in latest commit.

Comment (dhgreene) [reply ops/build_and_package.sh]: 🤦‍♂ I'm really glad you're looking at this.  Fixed in latest commit.

**Comment by msnook** [comment]
From a packaging perspective, I'd like to make sure the AMI is configured properly.  Can you deploy your feature branch to `dev` to ensure the swagger components are rendered appropriately in AWS?

File: ssas/service/main/main.go [hunk 3]
```diff
@@ -1,3 +1,29 @@
+/*
+ Package main System-to-System Authentication Service
+
+ The System-to-System Authentication Service (SSAS) enables one software system to authenticate and authorize another software system. In this model, the Systems act automatically, independent of a human user identity. Human users are involved only to administer the Service, including establishing the identities and privileges of participating systems.
+
+ For more details see our repository readme and Postman tests:
+ - https://github.com/CMSgov/bcda-ssas-app
+ - https://github.com/CMSgov/bcda-ssas-app/tree/master/test/postman_test
+
+ If you have a Client ID and Secret you can use this page to explore the API.  To do this, click the green "Authorize" button below and enter your Client ID and secret in the Basic Authentication username and password boxes.
```
Comment (tbellj) [inline ssas/service/main/main.go]: not sure how to do this but should we update the username and password boxes text to say Client ID and Client Secret?? Unsure about the level of effort tho

Comment (dhgreene) [reply ssas/service/main/main.go]: We get the functionality by saying we use Basic authentication.  I'm pretty sure we can't update the username/password boxes without mucking around in `go-swagger` source.

It _would_ be really, really nice, though.

**Comment by msnook** [comment]
You might need something similar to this: 
https://github.com/CMSgov/bcda-app/blob/master/Makefile#L4

File: ssas/service/main/main.go (hunk 3 above)
Comment (dhgreene) [reply ssas/service/main/main.go]: OK, accomplished with a string search/replace.  Presumably we don't update the `.js` bundles very often.  😄

**Pull request merged**